src/ImageGallery.jsx

```diff
@@ -847,19 +936,26 @@ export default class ImageGallery extends React.Component {
     }
   }
 
-
   handleScreenChange() {
-    /*
-      handles screen change events that the browser triggers e.g. esc key
-    */
+    /**
+     * handles screen change events that the browser triggers (e.g. esc key),
+     * but only if the current instance is the gallery shown in fullscreen mode.
+     * This is especially useful, when multiple galleries are used on a single
+     * page.
+     */
     const { onScreenChange } = this.props;
-    const fullScreenElement = document.fullscreenElement
-      || document.msFullscreenElement
-      || document.mozFullScreenElement
-      || document.webkitFullscreenElement;
-
-    if (onScreenChange) onScreenChange(fullScreenElement);
-    this.setState({ isFullscreen: !!fullScreenElement });
+    const fullScreenElement = ImageGallery.getFullScreenElement();
+    const isFullscreen = this.imageGallery.current === fullScreenElement;
+
+    this.setState({ isFullscreen }, () => {
+      if (onScreenChange) {
+        onScreenChange(
+          this.state.isFullscreen && fullScreenElement
```

Comment:
is the && fullScreenElement necessary, isn't it already either the element or undefined?

Reply:
Resolved here: https://github.com/lovelysystems/react-image-gallery/pull/1/commits/7b23c2d2579ecc27194f88f9fa3e3459c7e99b41#diff-eabaad0cc8751d0947757ba5349ae171R956-R968